### DISCUSSION TITLE
Configuring Authorization Bearer token

### DISCUSSION THREAD

**Discussion by YurySuslovScalyr** [jaegertracing/jaeger#4368] [Q&A]
How can I configure the Authorization: Bearer xxx token to be forwarded to storage backend that I am writing? I am using jaeger-all-in-one Docker container for testing purposes. I didn't find any relevant configuration either in Query, or in UI.

**Comment by yurishkuro** [comment]
https://github.com/jaegertracing/jaeger/discussions/4367#discussioncomment-5528275

**Comment by albertteoh** [comment]
@YurySuslovScalyr does this work: `--prometheus.token-file=<path to file where bearer token stored>`? (I haven't tried it myself)

**Reply by YurySuslovScalyr** [reply]
@albertteoh No, unfortunately it does not. I've tried prometheus, es & es-archive tokens, with a command like this:

```
docker run \
  -v /Users/yurys/jaeger:/tmp \
  --rm --name jaeger \
  -e SPAN_STORAGE_TYPE=grpc-plugin \
  -e GRPC_STORAGE_SERVER="host.docker.internal:4774" \
  -e QUERY_BEARER_TOKEN_PROPAGATION=true \
  -e PROMETHEUS_TOKEN_FILE="/tmp/token" \
  -e ES_ARCHIVE_TOKEN_FILE="/tmp/token" \
  -e ES_TOKEN_FILE="/tmp/token" \
  -p 16686:16686 \
  jaegertracing/all-in-one:1.43
```

I've checked that the token file is there and is accessible.

**Reply by albertteoh** [reply]
@YurySuslovScalyr can you please try using `jaegertracing/all-in-one:1.44` ([additional context](https://github.com/jaegertracing/jaeger/discussions/4367#discussioncomment-5551621))?

**Reply by YurySuslovScalyr** [reply]
@albertteoh still does not work unfortunately, I don't see any additional headers when debugging my grpc storage server.

**Reply by albertteoh** [reply]
Ah, sorry @YurySuslovScalyr, I assumed you wanted to pass the bearer token to metrics storage. 

For span storage, ES_TOKEN_FILE will work only for elastic search storage backends. I'm not aware of a similar mechanism for configuring bearer tokens to be passed to grpc storage.
